Documentation updates (#10178)
Homebrew/brew: docs/Maintainer-Guidelines.md

  [pip](https://pip.pypa.io/en/stable/).
- Ensure that any dependencies are accurate and minimal. We don't need to
  support every possible optional feature for the software.

> - Use the GitHub squash & merge or rebase & merge workflows where bottles aren't required.

Author: If it's still not clear, let's fix it. I think what you suggested works well—gives just the right amount of information while still being readable.

How about

```suggestion
- When bottles aren't required or affected, use the GitHub squash & merge workflow for a single-formula PR or rebase & merge workflow for a multiple-formulae PR. See [below](#merge-without-bottles-table) for more details.
```

I added the second `workflow` simply because I like the was it reads but I don't feel strongly about that.

I also added a link to the table to make it easy for someone who looks at that and is a little confused to get some more information.

Thought?

> - Use the GitHub squash & merge or rebase & merge workflows where bottles aren't required.

Reviewer: I was on the fence about the second "workflow" but I'm good with it (and the link to the table). Looks good to me 👍

### Merging, rebasing, cherry-picking

> Merging should be done in the `Homebrew/brew` repository to preserve history & GPG commit signing.

Reviewer: ```suggestion
Merging should be done in the `Homebrew/brew` repository to preserve history and GPG commit signing.
```

This is the only other thing I think needs to be changed. Everything else looks good to me at this point.

### Merging, rebasing, cherry-picking

Merging should be done in the `Homebrew/brew` repository to preserve history & GPG commit signing,

> and squash/merge or rebase/merge via GitHub should be used for formulae where those formulae

Reviewer: Reading over this again, I feel like this sentence has gotten a bit lengthy at this point. How do you feel about ending the sentence at “GPG commit signing” and starting a new sentence like “Squash/merge or rebase/merge”, etc. Besides that, this would allow us to reword the end of the previous sentence to “...preserve history and GPG commit signing, as the ampersand feels a bit strange to me here.

This is also the one instance where we use “squash/merge” instead of “squash & merge”, so there may be something to be said for standardizing on one style and using it throughout.

These are the only other things that stuck out to me and I’m good with everything else. If you think this is fine without additional changes, then I’m okay to approve it a little later when I have another free moment.

Author: Just pushed a new change to improve this. Let me know what you think.

![Flowchart for managing pull requests](assets/img/docs/managing-pull-requests.drawio.svg)

> Here are guidelines about when to use squash & merge versus rebase & merge. These options should only be used when bottles are not needed.

Reviewer: ```suggestion
Here are guidelines about when to use squash & merge versus rebase & merge. These options should only be used with PRs where bottles are not needed or affected.
```

As above, it may be beneficial to disambiguate the language around bottles here but there may be a better way of expressing this than what I've suggested.

Author: 👍 Like the added clarity about needed vs affected
